orca_detector/training.py

```diff
@@ -72,15 +75,40 @@ def train(model_name):
     print(f'Removed the following classes:\n{orca_params.REMOVE_CLASSES}')
     print(f'Mapped the following classes to "{orca_params.OTHER_CLASS}":\n' \
           f'{orca_params.OTHER_CLASSES}\n')
-    
 
     model = create_network(model_name, classes)
 
+    results_folder = os.path.join(orca_params.OUTPUT_PATH, model_name)
+    if not os.path.exists(results_folder):
+        os.makedirs(results_folder)
+
+    weight_path = os.path.join(results_folder, orca_params.BEST_WEIGHT_FILE_NAME)
```

Comment:
By using a hard-coded name, rather than a timestamped name (and symlink), we will overwrite this file every time we do a training run.  The weights we want to use for inference won't necessarily be from the most recently run experiment, so we should keep the best weights from each training run.  

In `orca_utils.save_model()` we're saving the weights file with val loss, val acc, and the run_timestamp "key" (which allows us to associate these weights with other artifacts like loss plots) in the filename.  I'm not sure if the Keras checkpoint callback will let you insert metrics like loss into the filename, but you should at least be able to append RUN_TIMESTAMP here in line 85 so we know which weights came from which run.  Then after the training has completed you can apply a symlink to "hide" the timestamp.

Once you're timestamping the best weights, I think that `orca_utils.save_model()` could basically be changed to be `orca_utils.save_model_config()`, outputting the json config of the model, but it no longer needs to save the weights.

Reply:
Agreed. I updated the save paths logic to use folders in the results/model name/time stamp/weights. I had some issues with the output structure of tensorboard, that was updated too. I like now how everything is grouped under the model name folder.